plugin/trino-oracle/src/test/java/io/trino/plugin/oracle/TestOracleConnectorTest.java

```diff
@@ -83,4 +90,56 @@ protected SqlExecutor onRemoteDatabase()
     {
         return oracleServer::execute;
     }
+
+    @Test
+    public void testPushdownJoinWithLongNameSucceeds()
+    {
+        tryCleanupTemporaryTable();
+        try {
+            String baseColumnName = "test_pushdown_" + randomNameSuffix();
+            String validColumnName = baseColumnName + "z".repeat(MAX_CHARS_COLUMN_ALIAS - baseColumnName.length());
+
+            assertUpdate(format("""
+                    CREATE TABLE orders_1 as
+                        SELECT orderkey as %s,
+                            custkey,
+                            orderstatus,
+                            totalprice,
+                            orderdate,
+                            orderpriority,
+                            clerk,
+                            shippriority,
+                            comment
+                        FROM orders
+                    """, validColumnName), "VALUES 15000");
+
+            Session session = Session.builder(getSession())
+                    .setCatalogSessionProperty("oracle", JOIN_PUSHDOWN_ENABLED, "true")
+                    .build();
+            assertQuery(session,
+                    format("""
+                            SELECT c.custkey, o.%s, n.nationkey
+                                 FROM orders_1 o JOIN customer c ON c.custkey = o.custkey
+                                 JOIN nation n ON c.nationkey = n.nationkey
+                            """, validColumnName),
+                    """
+                                SELECT c.custkey, o.orderkey, n.nationkey
+                                FROM orders o JOIN customer c ON c.custkey = o.custkey
+                                JOIN nation n ON c.nationkey = n.nationkey
+                            """);
+        }
+        finally {
+            tryCleanupTemporaryTable();
+        }
+    }
```

Comment:
Also while this test by itself is sufficient to prove that the aliasing logic works it doesn't prove that we chose an optimal value for the max identifier length.

IMO we should have another test to verify the length we chose:

- Try create a column with length = max_length - must succeed
- Try create a column with length = max_length + 1 - must fail in expected manner

@kokosing the reason why i believe this is useful is because for example we can use 30 as the max globally but it un-necessarily loses information in connectors which can actually support longer identifiers making reading explain plans harder than needed.